ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/Node.hs

```diff
@@ -772,6 +894,31 @@ protocolInfoCardano protocolParamsByron@ProtocolParamsByron {
           Comp maxTxCapacityOverridesAlonzo  :*
           Nil
 
+    blockForgingShelleyPraosBased :: m [OptNP 'False (BlockForging m) (CardanoEras c)]
+    blockForgingShelleyPraosBased = do
+        shelleyBased <-
+          traverse
+            (\creds -> Praos.praosSharedBlockForging
```

Comment:
While working on how to avoid `optAppendSkip`, I've come to the conclusion that its wrong to call both `shelleySharedBlockForging` and `praosSharedBlockForging` on the same leader credentials.

Both of those functions allocate a hot key for the credentials. I think we instead only want one hot key per credentials.

I'm going to continue preparing a change here, but it will have slightly broader scope that we had planned. (I could maybe most of it out into a predecessor PR if we want.)

Reply:
I just now pushed up commit 16501e07ee16bc40d60bd40b475381f76ffd46a4 to branch `nfrisby/praos-protocol` with what I think is a (slightly heavyhanded?) fix. Like I said, it could be broken apart into a separate PR before this one without too much trouble.

Reply:
@nc6 in particular, is it right to `assert` that those couple TPraos/Praos protocol parameters will be equivalent?

Reply:
Sorry, somehow missed this thread. I've commented in your other branch, but in particular, yes, the assertions are sound and it seems a definite simplification!

Reply:
On a call: we decided Clarke will cherry-pick and/or squash from my branch.